src/app-layer-smb.c

```diff
@@ -36,11 +39,145 @@ static SuricataFileContext sfc = { &sbcfg };
 static void SMBParserRegisterTests(void);
 #endif
 
+static IPPairStorageId g_ippair_smb_multi_id = { .id = -1 };
+
+#define GUID_LEN 16
+
+typedef struct SmbMultiIppairKey {
+    uint8_t guid[GUID_LEN];
+    uint64_t filesize;
+    uint8_t *filename;
+    uint16_t filename_len;
+} SmbMultiIppairKey;
+
+void SmbMultiSetFileSize(const Flow *f, const uint8_t *guid, uint64_t eof, const uint8_t *filename,
+        uint16_t name_len)
+{
+    IPPair *ipp;
+    Address ip_src, ip_dst;
+
+    if (GetFlowAddresses(f, &ip_src, &ip_dst) == -1)
+        return;
+    ipp = IPPairGetIPPairFromHash(&ip_src, &ip_dst);
+    if (ipp == NULL)
+        return;
+
+    SmbMultiIppairKey *key = IPPairGetStorageById(ipp, g_ippair_smb_multi_id);
+
+    if (key) {
+        // already a key
+        if (memcmp(guid, key->guid, GUID_LEN) == 0) {
+            // same guid, update only eof
+            if (eof > key->filesize) {
+                key->filesize = eof;
+            }
+            if (name_len > 0 && key->filename == NULL) {
+                key->filename = SCMalloc(name_len);
+                if (key->filename == NULL) {
+                    SCFree(key);
+                    goto end;
+                }
+                key->filename_len = name_len;
+                memcpy(key->filename, filename, name_len);
+            }
+            goto end;
+        } else {
+            // different guid : keep old one TODOsmbmulti6 : handle multimulti ?
```

Comment:
Do we want to handle multiple files, each over multiple streams ?

Reply:
For one ip pair

Reply:
I don't know for sure, but I would be quite surprised if there could just be a single multi stream file transfer per ip-pair. So this probably has to be a list.

Also wondering if such a list could be a Rust implementation. Guess it would need a void ptr in the key.

Reply:
Solved by dropping use of ippair and only using global hash table as for HTTP ranges (see below)